client/src/communication/http.rs

```diff
@@ -0,0 +1,97 @@
+use std::time::Duration;
+use reqwest::Client;
+use reqwest::header::CONTENT_TYPE;
+
+const TIMEOUT: Duration = Duration::from_secs(2);
+
+const CONTENT_HEADER_VALUE: &'static str = "application/json";
+const AUTHORIZATION_HEADER: &'static str = "Authorization";
+
+/// Sends a post request to the control server with a JSON String as body
+///
+/// # Returns
+/// The response from the request as a String
+///
+/// # Errors
+/// This function fails if there are issues sending the request, if 200 is not returned from
+/// the server or if parsing of the request response text fails
+pub async fn post_request(body: String,
+                          url: String,
+                          auth_token: &String) -> Result<String, anyhow::Error> {
+    let client = Client::builder()
+        .timeout(TIMEOUT)
+        .build()?;
+    let res = client.post(url)
+        .header(CONTENT_TYPE, CONTENT_HEADER_VALUE)
```

Comment:
Nice to see that the ContentType header is included for HTTP(s) requests, and not in UDP requests! The HTTP(s) requests go directly to the Flask control server, and therefore need the header to know how to parse the content. The UDP server should append them though, so it isn't needed there.

Reply:
Pure luck that I forgot that one!